allennlp/models/encoder_decoders/wikitables_semantic_parser.py

```diff
@@ -147,41 +166,72 @@ def forward(self,  # type: ignore
+        # (batch_size, num_entities, num_entities)
+        neighbor_indices = self._get_neighbor_indexes(world, num_entities, encoded_table)
+        # (batch_size, num_entities, num_entities, embedding_dim)
+        embedded_neighbors = util.batched_index_select(encoded_table, neighbor_indices)
+        neighbor_mask = util.get_text_field_mask({'ignored': neighbor_indices},
+                                                 num_wrapping_dims=1).float()
+        # (batch_size, num_entities, embedding_dim)
+        embedded_neighbors = self._neighbor_encoder(embedded_neighbors, neighbor_mask)
+
+        # (batch_size, num_entities, num_types)
+        type_vector, entity_types = self._get_type_vector(world, num_entities, encoded_table)
+
+        type_linear_combination = self._type_params(type_vector.float())
+        neighbor_linear_combination = self._neighbor_params(embedded_neighbors.float())
+        # (batch_size, num_entities, embedding_dim)
+        entity_embeddings = torch.nn.functional.tanh(type_linear_combination + neighbor_linear_combination)
+
+        # compute entity and question word similarity through a dot product
+        sim_dot_prod = torch.bmm(embedded_table.view(batch_size, -1, self._embedding_dim),
+                                 torch.transpose(embedded_question, 1, 2))
+        sim_dot_prod = sim_dot_prod.view(batch_size, num_entities, num_entity_tokens, num_question_tokens)
+        # Note the 0th index is accessed below since max returns a tuple of a float and long tensor.
+        # (batch_size, num_entities, num_question_tokens)
+        max_sim = torch.max(sim_dot_prod, 2)[0]
         # (batch_size, num_entities, num_question_tokens, num_features)
-        linking_features = table['linking']  # pylint: disable=unused-variable
-        # TODO(mattg): multiply this by the feature weights, use this and the text embedding above
-        # to compute a linking scores.
+        linking_features = table['linking']
+        linking_scores = max_sim + self._linking_params(linking_features.float()).squeeze(3)
 
-        # TODO(mattg): we need to actually compute this with the stuff mentioned above.  This
-        # represents how well each question token matches table entities.  I'm stubbing this out
-        # for now, so that I have a variable to pass to the decoder, because we need this there.
-        # (batch_size, num_entities, num_question_tokens)
-        num_entities = embedded_table_text.size(1)
-        num_question_tokens = embedded_input.size(1)
-        linking_scores: torch.FloatTensor = Variable(torch.ones(batch_size, num_entities, num_question_tokens))
+        # (batch_size, num_entities+1, num_question_tokens)
+        softmax_scores = self._get_prob_entity(world, linking_scores, batch_size, num_entities,
+                                               num_question_tokens, encoded_table)
+
+        null_entity_embedding = Variable(entity_embeddings.data.new(torch.zeros(batch_size, 1,
```

Comment:
You can just do `Variable(entity_embeddings.data.new(batch_size, 1, self._embedding_dim).fill_(0))`.